code/object/objectdock.cpp

```diff
@@ -337,6 +338,68 @@ float dock_calc_docked_speed(object *objp)
 	return dfi.maintained_variables.float_value;
 }
 
+void dock_calc_total_moi(matrix* dest, object* objp, vec3d *center)
+{
+	Assertion(dest != nullptr, "dock_calc_total_moi invalid dest");
+	Assertion(objp != nullptr, "dock_calc_total_moi invalid objp");
+
+	matrix accum;
+	vm_mat_zero(&accum);
+
+	dock_function_info dfi;
+	dfi.parameter_variables.vecp_value = center;
+	dfi.maintained_variables.matrix_value = dest;
+
+	dock_evaluate_all_docked_objects(objp, &dfi, dock_calc_inv_total_moi_helper);
+}
+
+void dock_whack_all_docked_objects(vec3d* force, vec3d* world_hit_pos, object* objp)
```

Comment:
Ack, no no no.  You don't whack all the ships, you whack *one* ship using the aggregate mass and aggregate MOI.